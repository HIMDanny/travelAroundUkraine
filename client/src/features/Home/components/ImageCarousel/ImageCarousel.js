/* eslint-disable import/no-unresolved */
import React, { useEffect, useState } from 'react';

import Carousel from 'react-material-ui-carousel';

const ImageCarousel = () => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    (async () => {
      const data = await fetch('./heroSectionImages.json').then((res) => res.json());
      setImages(data);
    })();
  }, []);

  return (
    <Carousel
      height="100vh"
      interval={6500}
      duration={2500}
      stopAutoPlayOnHover={false}
      navButtonsAlwaysInvisible
      indicators={false}
    >
      {images.map((url) => (
        <img src={url} alt="Ukraine" key={url} />
      ))}
    </Carousel>
  );
};

export default ImageCarousel;
