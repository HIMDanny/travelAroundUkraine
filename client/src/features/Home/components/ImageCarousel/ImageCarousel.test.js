import { render } from '@testing-library/react';
import ImageCarousel from './ImageCarousel';

describe('Image Carousel Snapshot test', () => {
  test('should ImageCarousel match snapshot', () => {
    const { asFragment } = render(<ImageCarousel />);
    expect(asFragment()).toMatchSnapshot();
  });
});
