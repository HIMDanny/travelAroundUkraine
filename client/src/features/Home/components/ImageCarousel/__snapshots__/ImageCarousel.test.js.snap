// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Carousel Snapshot test should ImageCarousel match snapshot 1`] = `
<DocumentFragment>
  <div
    class=" css-ohwg9z"
  >
    <div
      class="css-1f8sh1y"
      style="height: 100vh;"
    />
  </div>
</DocumentFragment>
`;
